cfgov/v1/models/base.py

```diff
@@ -217,6 +220,29 @@ def specific_categories_query(block, parent_slug):
         return {search_type: queryset for search_type, queryset in
                 related.items() if queryset}
 
+    def flush(self):
+        if not settings.ENABLE_AKAMAI_CACHE_PURGE:
+            return False
+        auth, fast_purge_url = get_akamai_credentials()
+        page_relative_path = self.url_path.replace("cfgov/", "")
```

Comment:
It should work with any pages that are published under a site root. See [docs](http://docs.wagtail.io/en/v1.9/reference/pages/model_reference.html#wagtail.wagtailcore.models.Page.get_site) and [source code](https://github.com/wagtail/wagtail/blob/master/wagtail/wagtailcore/models.py#L853). It's supposed to return null if the site doesn't exist, so I'm curious where your errors are coming from. It seems reasonable to me to skip the cache step for pages that aren't published, e.g.

```py
site = self.get_site()

if not site:
    return

# do something with self.relative_url(site) ...
```